proof/refine/ARM/Syscall_R.thy

```diff
@@ -920,22 +928,36 @@ lemma threadSet_not_curthread_ct_domain:
   apply clarsimp
   done
 
+lemma tcbSchedDequeue_not_in_queue:
+  "\<And>p. \<lbrace>Invariants_H.valid_queues and tcb_at' t and valid_objs'\<rbrace> tcbSchedDequeue t
+   \<lbrace>\<lambda>rv s. t \<notin> set (ksReadyQueues s p)\<rbrace>"
```

Comment:
Do we need `tcb_at’ t` and `valid_objs’` here?

Reply:
I cut and paste from an old proof, will optimise it now.